django/db/models/query.py

```diff
@@ -1208,17 +1236,107 @@ def _raw_delete(self, using):
 
     _raw_delete.alters_data = True
 
-    def update(self, **kwargs):
+    # Existing signature with keyword only updates.
+    @overload
+    def update(self, **updates) -> int: ...
+
+    # New signature with positional only updates to preserve backward
+    # compatibility with update(updates={"something"}) for models with
+    # an "updates" field.
+    @overload
+    def update(self, updates: dict, /) -> int: ...
+
+    # New signature with positional only updates and returning support which
+    # also allows paves the way for other options in the future if needs be.
+    @overload
+    def update(
+        self, updates: dict, *, returning: Literal[bool] | Iterable[str]
+    ) -> list: ...
+
+    # New signature with positional only updates and returning support which
+    # also allows paves the way for other options in the future if needs be.
+    @overload
+    def update(
+        self, updates: dict, *, returning_values: Literal[bool] | Iterable[str]
+    ) -> list[dict]: ...
+
+    # New signature with positional only updates and returning support which
+    # also allows paves the way for other options in the future if needs be.
+    @overload
+    def update(
+        self, updates: dict, *, returning_values_list: Literal[bool] | Iterable[str]
+    ) -> list[tuple]: ...
+
```

Comment:
These are only added to explain the signature handling bellow better before the docs are adjusted.